#!/usr/bin/env python
PACKAGE = "mercury_trajectory"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("curve_movement", bool_t, 0, "Curve Movement", False)

gen.add("linear_axes", str_t, 0, "Axes of linear-x when you want to go forward", "linear_axes")
gen.add("angular_axes", str_t, 0, "Axes of angular-z when you want to go forward", "angular_axes")
gen.add("axes_threshold", double_t, 0, "Threshold of axes to button", 0.5, 0.0, 1.0)

gen.add("btn_turn_left", str_t, 0, "Button of angular-z when you want to turn left", "btn_turn_left")
gen.add("btn_turn_right", str_t, 0, "Button of angular-z when you want to turn left", "btn_turn_right")
gen.add("btn_speed_decr", str_t, 0, "Speed decrease button", "btn_speed_decr")
gen.add("btn_speed_ecnr", str_t, 0, "Speed encrease button", "btn_speed_ecnr")
gen.add("btn_front_flipper_up", str_t, 0, "Front flipper up button", "btn_front_flipper_up")
gen.add("btn_front_flipper_down", str_t, 0, "Front flipper down button", "btn_front_flipper_down")
gen.add("btn_rear_flipper_up", str_t, 0, "Rear flipper up button", "btn_rear_flipper_up")
gen.add("btn_rear_flipper_down", str_t, 0, "Rear flipper down button", "btn_rear_flipper_down")

exit(gen.generate(PACKAGE, "mercury_trajectory", "Trajectory"))